render/render.c

```diff
@@ -0,0 +1,378 @@
+#include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <EGL/egl.h>
+#include <EGL/eglext.h>
+#include <GLES2/gl2.h>
+#include <GLES2/gl2ext.h>
+
+#include <wlr/backend.h>
+#include <wlr/types/wlr_output.h>
+#include <wlr/util/log.h>
+#include "render/render.h"
+#include "render/glapi.h"
+
+static const struct format formats[] = {
+	{
+		.wl_fmt = WL_SHM_FORMAT_ARGB8888,
+		.gl_fmt = GL_BGRA_EXT,
+		.gl_type = GL_UNSIGNED_BYTE,
```

Comment:
A little comment "WL_SHM_FORMAT_ARGB8888 is in little-endian, GL_BGRA_EXT is in big-endian" might help newcomers. I was rather confused the first time I read this code, and didn't understand it.